### PR TITLE
Fix the filters of multiple tables

### PR DESCRIPTION
**Pull request by priyesh777**
Addresses:- 
https://github.com/idmc-labs/Helix2.0/issues/200#issue-950392091

Depends on:-
https://github.com/idmc-labs/helix-server/pull/194#issue-684322199
https://github.com/idmc-labs/helix-server/pull/198#issue-694291677

## Changes
- Add addtional filters in country, crises, events and extraction

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

**Review by tnagorra — CHANGES_REQUESTED** [review]
Doesn't work with develop branch.
Please add the dependency with the server.

File: src/views/Extraction/NewExtractionFilters/index.tsx [hunk 1]
```diff
@@ -434,6 +481,44 @@ function NewExtractionFilters(props: NewExtractionFiltersProps) {
...
+                />
+                {/* <TextInput
+                    className={styles.input}
+                    icons={<IoIosSearch />}
+                    label="Year"
+                    name="year"
+                    value={value.year}
+                    onChange={onValueChange}
+                    placeholder="Search"
+                /> */}
```
Comment (tnagorra) [inline src/views/Extraction/NewExtractionFilters/index.tsx:525]: We do not need this! Remove this!

File: src/views/Extraction/NewExtractionFilters/index.tsx [hunk 2]
```diff
@@ -434,6 +481,44 @@ function NewExtractionFilters(props: NewExtractionFiltersProps) {
...
+                    onChange={onValueChange}
+                    error={error?.fields?.filterEntryTags?.$internal}
+                    disabled={disabled}
+                    onOptionsChange={setTerms}
+                />
+                <MultiSelectInput
+                    options={data?.displacementType?.enumValues}
+                    keySelector={enumKeySelector}
+                    labelSelector={enumLabelSelector}
+                    label="Rural /Urban"
```
Comment (tnagorra) [inline src/views/Extraction/NewExtractionFilters/index.tsx:502]:
```suggestion
                    label="Rural / Urban"
```

File: src/views/Extraction/NewExtractionFilters/index.tsx [hunk 3]
```diff
@@ -235,6 +258,7 @@ function NewExtractionFilters(props: NewExtractionFiltersProps) {
                     filterEventCrises: otherAttrs.filterEventCrises?.map((cr) => cr.id),
                     filterFigureCategories: otherAttrs.filterFigureCategories?.map((fc) => fc.id),
                     filterEntryTags: otherAttrs.filterEntryTags?.map((ft) => ft.id),
+                    filterFigureTerms: otherAttrs.filterFigureTerms?.map((Fterms) => Fterms.id),
```
Comment (tnagorra) [inline src/views/Extraction/NewExtractionFilters/index.tsx:265]:
```suggestion
                    filterFigureTerms: otherAttrs.filterFigureTerms?.map((figureTerms) => figureTerms.id),
```

File: src/views/Events/EventsFilter/index.tsx [hunk 4]
```diff
@@ -169,6 +205,14 @@ function EventsFilter(props: EventsFilterProps) {
                         onChange={onValueChange}
                         error={error?.fields?.countries?.$internal}
                     />
+                    <TextInput
+                        className={styles.input}
+                        label="Glide Number"
+                        name="glideNumber_Icontains"
+                        value={value.glideNumber_Icontains}
+                        onChange={onValueChange}
+                        placeholder="Glide No."
```
Comment (tnagorra) [inline src/views/Events/EventsFilter/index.tsx:214]:
```suggestion
                        placeholder="Glide Number"
```

File: src/views/Events/EventsFilter/index.tsx [hunk 5]
```diff
@@ -134,6 +147,29 @@ function EventsFilter(props: EventsFilterProps) {
                         onChange={onValueChange}
                         placeholder="Search"
                     />
+                    {/*
+                    (FIX: This has been added as per requirement
+                         but the respective query is not updated)
```
Comment (tnagorra) [inline src/views/Events/EventsFilter/index.tsx:152]:
```suggestion
                    (FIXME: This has been added as per requirement
                         but the respective query is not updated)
```
What do you mean?

File: src/components/tables/CommunicationAndPartners/CommunicationTable/CommunicationFilter/index.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+const defaultFormValues: PartialForm<FormType> = {
+    subject: undefined,
+};
+
+interface ContactsFilterProps {
+    className?: string;
+    onFilterChange: (value: PurgeNull<CommunicationListQueryVariables>) => void;
+}
+
+function ContactsFilter(props: ContactsFilterProps) {
```
Comment (tnagorra) [inline src/components/tables/CommunicationAndPartners/CommunicationTable/CommunicationFilter/index.tsx:41]:
```suggestion
function CommunicationFilter(props: ContactsFilterProps) {
```

File: src/components/tables/CommunicationAndPartners/CommunicationTable/CommunicationFilter/index.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    fields: (): FormSchemaFields => ({
+        subject: [],
+    }),
+};
+
+const defaultFormValues: PartialForm<FormType> = {
+    subject: undefined,
+};
+
+interface ContactsFilterProps {
```
Comment (tnagorra) [inline src/components/tables/CommunicationAndPartners/CommunicationTable/CommunicationFilter/index.tsx:36]:
```suggestion
interface CommunicationFilterProps {
```

File: src/components/selections/FigureTermMultiSelectInput/index.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
+import React, { useMemo, useState } from 'react';
```
Comment (tnagorra) [inline src/components/selections/FigureTermMultiSelectInput/index.tsx:1]: Did we need this component?

File: src/views/Crises/CrisesFilter/index.tsx [hunk 9]
```diff
@@ -45,18 +46,35 @@ const CRISIS_OPTIONS = gql`
...
+const EVENT_OPTIONS = gql`
+    query EventOptionsForFilter {
+        eventType: __type(name: "CRISIS_TYPE") {
+            enumValues {
+                name
+                description
+            }
+        }
+    }
+`;
```
Comment (tnagorra) [inline src/views/Crises/CrisesFilter/index.tsx:58]: You don't have to create two different query. You can just add all of these on CRISIS_OPTIONS

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**